src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -253,14 +256,61 @@ public RecurringSession withStartDate(SessionDate newSessionDate) {
     }
 
     /**
-     * Creates a new {@RecurringSession} with the new {@code newLastSessionDate}.
+     * Creates a new {@code RecurringSession} with the new {@code newLastSessionDate}.
      * @param newLastSessionDate the new last session date.
      */
     public RecurringSession withLastSessionDate(SessionDate newLastSessionDate) {
         return new RecurringSession(getSessionDate(), getDuration(), getSubject(),
                 getFee(), getInterval(), newLastSessionDate);
     }
 
+    /**
+     * Removes a single {@code Session} with {@code sessionDate}.
+     * Splits the remaining {@code RecurringSession} into two {@code RecurringSession}
+     */
+    public void removeSessionInRecurringSession(Index sessionIndex,
```

Comment:
Haha, not sure if it is possible to abstract further. But then as attached below, when grading our project, having reasonable SLAP methods and not having deeply nested method is part of the consideration in grading. So it might be better to try to split up the code and attempt to reduce the nesting here also.
![image](https://user-images.githubusercontent.com/24240408/114002691-aae7df00-988f-11eb-9a3c-b1bbe378b6fb.png)

https://nus-cs2103-ay2021s2.github.io/website/admin/tp-grading.html#2-project-grading-implementation-15-marks

Reply:
I agree with you too! Made some changes, thank you!